src/win/fs.c

```diff
@@ -1313,41 +1276,47 @@ static void fs__ftruncate(uv_fs_t* req) {
 
 
 static void fs__sendfile(uv_fs_t* req) {
-  int fd_in = req->file.fd, fd_out = req->fs.info.fd_out;
+  HANDLE fd_in = req->file.hFile;
+  HANDLE fd_out = req->fs.info.hFile_out;
   size_t length = req->fs.info.bufsml[0].len;
   int64_t offset = req->fs.info.offset;
-  const size_t max_buf_size = 65536;
+  const size_t max_buf_size = 65535; /* maximum size supported by pipe write operations across a network */
   size_t buf_size = length < max_buf_size ? length : max_buf_size;
-  int n, result = 0;
-  int64_t result_offset = 0;
+  int result = 0;
+  BOOL io_success = 1;
   char* buf = (char*) uv__malloc(buf_size);
   if (!buf) {
     uv_fatal_error(ERROR_OUTOFMEMORY, "uv__malloc");
   }
 
   if (offset != -1) {
-    result_offset = _lseeki64(fd_in, offset, SEEK_SET);
+    LARGE_INTEGER offset_u;
+    offset_u.QuadPart = offset;
+    io_success = SetFilePointerEx(fd_in, offset_u, NULL, FILE_BEGIN);
   }
 
-  if (result_offset == -1) {
+  if (!io_success) {
     result = -1;
   } else {
     while (length > 0) {
-      n = _read(fd_in, buf, length < buf_size ? length : buf_size);
-      if (n == 0) {
-        break;
-      } else if (n == -1) {
+      DWORD n, n_out;
+      io_success = ReadFile(fd_in, buf, length < buf_size ? length : buf_size, &n, NULL);
+      if (!io_success) {
         result = -1;
         break;
       }
+      else if (n == 0) {
+        break;
+      }
 
       length -= n;
 
-      n = _write(fd_out, buf, n);
-      if (n == -1) {
+      io_success = WriteFile(fd_out, buf, n, &n_out, NULL);
+      if (!io_success) {
         result = -1;
         break;
       }
+      /* hopefully n == n_out here (e.g. that fd_out wasn't opened non-blocking mode) */
```

Comment:
This is just documenting an existing potential deficiency (`_write` is just a thin wrapper for `WriteFile`, although I shudder to think of what this code used to do if someone accidentally opened a fd with `_O_TEXT `). I don't think anyone is passing in a pipe opened in non-blocking mode intentionally.